src/Components/CreateImageWizardV2/CreateImageWizard.tsx

```diff
@@ -139,6 +142,10 @@ const CreateImageWizard = ({ startStepIndex = 1 }: CreateImageWizardProps) => {
   const blueprintName = useAppSelector(selectBlueprintName);
   const blueprintDescription = useAppSelector(selectBlueprintDescription);
   const activationKey = useAppSelector(selectActivationKey);
+  const snapshotDate = useAppSelector(selectSnapshotDate);
+  const useLatest = useAppSelector(selectUseLatest);
+
+  const snapshotStepRequiresChoice = !useLatest && !snapshotDate;
```

Comment:
I'll take a look!

Reply:
Hmmm, I don't think his functionality matches my own, we may want to look at something a bit more cohesive for both, as I think a better pattern is to stop the user from progressing should they change a required field.

The current implementation of your input "details" input is not triggered "onChange" but rather onBlur, which would allow the user to click away. 

My implementation prevents the movement to the next step altogether. I think a fusion of these two approaches would be ideal. 

An input validator that changes on the "onChange" handler, marks an error in the field as @ezr-ondrej  PR accomplishes, but also prevents movement to the next step. 

<img width="677" alt="Screenshot 2024-04-22 at 7 51 14 PM" src="https://github.com/osbuild/image-builder-frontend/assets/38083295/0a32ed11-ca06-4cd0-9f39-41234ebd7fc1">
<img width="616" alt="Screenshot 2024-04-22 at 7 52 03 PM" src="https://github.com/osbuild/image-builder-frontend/assets/38083295/3558b1ce-99d4-48fd-bc28-04570e99d834">

I also noticed a bug with the details step due to that functionality. 
If navigate to the details step during edit, I cannot "step" as the "onBlur" event hasn't been fired yet.:
<img width="593" alt="Screenshot 2024-04-22 at 7 53 39 PM" src="https://github.com/osbuild/image-builder-frontend/assets/38083295/9e8b4ff6-be0d-4987-b8ea-74d3d1e8df73">

Reply:
My PR allows to do both, we just need to consume the validation state here and use it for `disableNext`.
I don't think it's perfect, but it allows for validation to be handled in a single place.
The way we do it currently requires the validation to be done here and on the component.

Nice find on the bug tho, that's bit tricky on my approach.
Might actually be best to validate directly in the Redux action :thinking:

Reply:
I'll look into this, I think at the minimum we'd want to add a debounce to the input to allow for instant feedback on state.

Reply:
I'd love to team up and come up with a better way to validate.
That said, I'd not block this PR on it.